core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/HttpSignerCommon.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.spi;
+
+import java.time.Clock;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+
+/**
+ * Keeps common signing properties used by different signers.
+ */
```

Comment:
This makes sense. I was thinking about this a bit when I was working on some of the v4+v4a attributes, but decided we probably shouldn't do it for those attributes. It makes a lot of sense for the signing clock, though.

I think we should be really strict in the javadoc about this. It should be things that CAN NEVER be signer-specific. A clock is universal, it makes sense here. We need to write the javadoc in a way that keeps things like service signing name out of this. In my opinion, clock is probably the only thing that ever makes sense here, but who knows what the future will hold...

Reply:
My initial thought was this goes in the `HttpSigner` interface itself and specific signers javadoc whether they use this property or not. What do folks think about that?